tripal_chado/src/Plugin/Field/FieldFormatter/ChadoEmbeddedEntityFormatter.php

```diff
@@ -0,0 +1,355 @@
+<?php
+
+namespace Drupal\tripal_chado\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Entity\EntityDisplayRepositoryInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\tripal\TripalField\Attribute\TripalFieldFormatter;
+use Drupal\tripal_chado\TripalField\ChadoFormatterBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'embedded entity' formatter.
+ */
+#[TripalFieldFormatter(
+  id: 'embedded_entity_view',
+  label: new TranslatableMarkup('Rendered entity'),
+  description: new TranslatableMarkup('A chado embedded entity.'),
+  field_types: [
+    'chado_analysis_type_default',
+    'chado_assay_type_default',
+    'chado_biomaterial_type_default',
+    'chado_contact_by_role_type_default',
+    'chado_contact_type_default',
+    'chado_dbxref_type_default',
+    'chado_featuremap_type_default',
+    'chado_feature_type_default',
+    'chado_organism_type_default',
+    'chado_project_type_default',
+    'chado_protocol_type_default',
+    'chado_pub_type_default',
+    'chado_stock_type_default',
+    'chado_study_type_default',
+    'chado_synonym_type_default',
+  ],
+)]
+class ChadoEmbeddedEntityFormatter extends ChadoFormatterBase {
+
+  /**
+   * The number of times this formatter allows rendering the same entity.
+   *
+   * @var int
+   */
+  const RECURSIVE_RENDER_LIMIT = 20;
```

Comment:
I set up a recursive pair of fields, and I got project with contact with project with contact with project... and the limit did indeed work and eventually stopped it at 20, but at around 11 the field is displayed so compressed that only one or two characters appeared per line. So I am thinking this limit is too large. But I also wonder if we could even just set it to 1. In that case I have a project with embedded contact which in turn has an embedded project, and the recursion ends there. Is there any use case needing more than that?